CRM/Core/Session.php

```diff
@@ -152,10 +152,21 @@ public function initialize($isRead = FALSE) {
   /**
    * Resets the session store.
    *
-   * @param int $all
+   * @param int|string $mode
+   *   1: Default mode. Deletes the `CiviCRM` data from $_SESSION.
+   *   2: More invasive version of that. (somehow)
+   *   'partial': Less invasive. Preserve basic data (eg current user ID) from this session. Reset everything else.
```

Comment:
could the key be `retain_current_login` so it's clear what this particular partiality is for?

Reply:
@ufundo  Yeah, naming is tricky. The placement for this is fairly central (*which means it's likely to hang-around for sometime*), so you kind of want to get a good name. OTOH, it's also quite obscure, and (*per GL comments*) it's unclear if/why the functionality is really necessary -- so you don't want overthink the name.

Anyway, I've pushed an update to make it `keep_login` (compromise on shortness vs ambiguity).